x-pack/plugins/task_manager/server/lib/log_health_metrics.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '../../../../../src/core/server';
+import { HealthStatus } from '../monitoring';
+import { TaskManagerConfig } from '../config';
+import { MonitoredHealth } from '../routes/health';
+
+export function logHealthMetrics(
+  monitoredHealth: MonitoredHealth,
+  logger: Logger,
+  config: TaskManagerConfig
+) {
+  let contextMessage;
+
+  let logAsWarn = monitoredHealth.status === HealthStatus.Warning;
+  const logAsError = monitoredHealth.status === HealthStatus.Error;
+  const driftInSeconds = (monitoredHealth.stats.runtime?.value.drift.p99 ?? 0) / 1000;
+
+  if (driftInSeconds >= config.monitored_stats_warn_drift_in_seconds) {
+    contextMessage = `Detected drift of ${driftInSeconds}s`;
```

Comment:
is "drift" going to be good enough for this message?  Should we use something like "delayed task start" or similar?  Maybe we need to continue to reference the word "drift" here if it's used in other places in the health stats, but it feels a little ... unknown? ... to me.  Would "Detected delayed task start (drift) of ${...}..." be too much?

Should we also be printing the configured value?  It's possible that some folks would have access to the logs, but not the config, so wouldn't know what the configured value was.

Maybe it's too much to jam all this into the message WITH the monitored health object.  Could we do two messages, one with the health stats, but one before that indicating the drift was exceeded with the details?

Reply:
Good idea. I changed the message and it's now printing before the JSON dump